clang/lib/Headers/hlsl/hlsl_intrinsics.h

```diff
@@ -80,6 +80,34 @@ void asuint(double3, out uint3, out uint3);
 _HLSL_BUILTIN_ALIAS(__builtin_hlsl_elementwise_splitdouble)
 void asuint(double4, out uint4, out uint4);
 
+//===----------------------------------------------------------------------===//
+// asint16 builtins
+//===----------------------------------------------------------------------===//
+
+/// \fn int16_t asint16(T Val)
+/// \brief Interprets the bit pattern of x as an 16-bit integer.
+/// \param Val The input value.
+#ifdef __HLSL_ENABLE_16_BIT
+
+template <typename T, int N>
+constexpr __detail::enable_if_t<__detail::is_same<int16_t, T>::value ||
+                                    __detail::is_same<uint16_t, T>::value ||
+                                    __detail::is_same<half, T>::value,
+                                vector<int16_t, N>>
+asint16(vector < T, N > V) {
+  return __detail::bit_cast<int16_t, T, N>(V);
+}
+
+template <typename T>
+constexpr __detail::enable_if_t<__detail::is_same<int16_t, T>::value ||
```

Comment:
this looks correct now: https://hlsl.godbolt.org/z/hbds7vo8W, however your test cases should be breaking in `asint16-errors.hlsl`.

As mentioned before all those `erro: no matching function for call to 'bit_cast'` lines should now be ` error: no matching function for call to 'asint16'`